Wikipedia/Code/TalkPageFetcher.swift

```diff
@@ -115,19 +104,28 @@ class TalkPageFetcher: Fetcher {
         
         performTokenizedMediaWikiAPIPOST(to: siteURL, with: params, reattemptLoginOn401Response: false) { result, httpResponse, error in
             
-            if let error = error {
-                completion(.failure(error))
-            }
-            
-            if let resultError = result?["error"] as? [String: Any],
-               let info = resultError["info"] as? String {
-                completion(.failure(RequestError.api(info)))
-                return
-            }
-            
-            if let resultSuccess = result?["success"] as? [String: Any] {
-                completion(.success(resultSuccess))
-            }
+            self.performTalkPagePost(error, result, completion: completion)
         }
     }
+    
+    fileprivate func performTalkPagePost(_ error: Error?, _ result: [String : Any]?, completion: @escaping(Result<[AnyHashable: Any], Error>) -> Void) {
```

Comment:
Maybe rename this to something like `evaluateResponse(_ error: Error?, _ result: [String : Any]?...)`? `performTalkPagePost` makes me think this method will make another POST call.